components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/listener/IdentityClaimValueEncryptionListener.java

```diff
@@ -0,0 +1,290 @@
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.mgt.listener;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.core.util.CryptoException;
+import org.wso2.carbon.core.util.CryptoUtil;
+import org.wso2.carbon.identity.claim.metadata.mgt.exception.ClaimMetadataException;
+import org.wso2.carbon.identity.claim.metadata.mgt.model.LocalClaim;
+import org.wso2.carbon.identity.core.AbstractIdentityUserOperationEventListener;
+import org.wso2.carbon.identity.core.util.IdentityCoreConstants;
+import org.wso2.carbon.identity.core.util.IdentityTenantUtil;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import org.wso2.carbon.identity.mgt.internal.IdentityMgtServiceDataHolder;
+import org.wso2.carbon.user.core.UserStoreException;
+import org.wso2.carbon.user.core.UserStoreManager;
+import org.wso2.carbon.user.core.common.AbstractUserStoreManager;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class IdentityClaimValueEncryptionListener extends AbstractIdentityUserOperationEventListener {
+
+    private static final Log LOG = LogFactory.getLog(IdentityClaimValueEncryptionListener.class);
+
+    private static final String CLAIM_VALUE = "ClaimValue";
+    private static final String CLAIM_URI = "claimURI";
+
+    @Override
+    public int getExecutionOrderId() {
+
+        int orderId = getOrderId();
+        if (orderId != IdentityCoreConstants.EVENT_LISTENER_ORDER_ID) {
+            return orderId;
+        }
+        return 96;
+    }
+
+
+    @Override
+    public boolean doPreAddUser(String userName, Object credential, String[] roleList, Map<String, String> claims,
+                                      String profile, UserStoreManager userStoreManager) throws UserStoreException {
+
+        try {
+            if (!isEnable() || userStoreManager == null) {
+                return true;
+            }
+            updateClaimValues(claims, userStoreManager);
+            return true;
+        } catch (org.wso2.carbon.user.api.UserStoreException e) {
+            LOG.error("Error while retrieving tenant ID", e);
+            return false;
+        } catch (CryptoException e) {
+            return false;
+        }
+    }
+
+    @Override
+    public boolean doPreAddUserWithID(String userID, Object credential, String[] roleList, Map<String, String> claims,
```

Comment:
"try {
            if (!isEnable() || userStoreManager == null) {
                return true;
            }
            updateClaimValues(claims, userStoreManager);
            return true;
        } catch (org.wso2.carbon.user.api.UserStoreException e) {
            LOG.error("Error while retrieving tenant ID", e);
            return false;
        } catch (CryptoException e) {
            LOG.error("Error occurred while encrypting claim value", e);
            return false;
        }"

If this block keeps repeating in several methods, we could have call a single method with logic via other methods